compiler-core/src/format.rs

```diff
@@ -692,7 +692,9 @@ impl<'comments> Formatter<'comments> {
 
             UntypedExpr::TupleIndex { tuple, index, .. } => self.tuple_index(tuple, *index),
 
-            UntypedExpr::Negate { value, .. } => self.negate(value),
+            UntypedExpr::NegateInteger { value, .. } => self.negate_with("-", value),
```

Comment:
We had someone the other day confused by the behaviour of the code `--x`, let's make it clear there's no `--` operator in Gleam by adding spaces or blocks

Reply:
`NegateInteger`

Would `NegateInt` be consistent with the other change requests?
`Integer` currently only exists 5 times in the whole gleam-lang/gleam repo, `Int` 1161 times.

Reply:
So

```gleam
let a = -b
```

should format to

```gleam
let a = - b
```

or just the case of repeated negation,

```gleam
let a = --b
```

to one of

```gleam
let a = - { -b }
let a = - -b
```

Reply:
I guess?:

- `let a = -b` is fine and expected.
- `let a = - b` should format to `let a = -b`
- `let a = --b` should format to `let a = b`

Reply:
That's an interesting idea. Bool negation doesn't do that right now, maybe that could be a feature request?

I think blocks look the nicest, e.g. `--a` becomes `- { -a }`. The newest changes do that

Reply:
The formatter doesn't change the AST at all so we wouldn't collapse negation operators, even if they are doing something pointless.  Using a block sounds great